packages/fetchai/connections/gym/connection.py

```diff
@@ -49,47 +49,38 @@ def __init__(self, address: Address, gym_env: gym.Env):
         """Initialize a gym channel."""
         self.address = address
         self.gym_env = gym_env
-        self._lock = threading.Lock()
+        self._lock: Optional[asyncio.Lock] = None
 
-        self._queues = {}  # type: Dict[str, asyncio.Queue]
+        self._queue: Optional[asyncio.Queue] = None
+        self._threaded_pool: ThreadPoolExecutor = ThreadPoolExecutor(10)
```

Comment:
Any reason for a high default of 10? Should also be declared as constant on top or be made configurable.